packml_ros/src/packml_ros.cpp

```diff
@@ -362,4 +353,111 @@ bool PackmlRos::loadStats(packml_msgs::LoadStats::Request &req, packml_msgs::Loa
 
   return true;
 }
+
+bool PackmlRos::commandGuard(const int& command_int, bool& command_valid, bool& command_rtn)
+{
+  command_valid = true;
+  command_rtn = false;
+
+  switch (command_int)
+  {
+    case static_cast<int>(packml_sm::CmdEnum::ABORT):
+      command_rtn = sm_->abort();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::CLEAR):
+      command_rtn = sm_->clear();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::HOLD):
+      command_rtn = sm_->hold();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::RESET):
+      command_rtn = sm_->reset();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::START):
+      command_rtn = sm_->start();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::STOP):
+      command_rtn = sm_->stop();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::SUSPEND):
+      command_rtn = sm_->suspend();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::UNHOLD):
+      command_rtn = sm_->unhold();
+      break;
+    case static_cast<int>(packml_sm::CmdEnum::UNSUSPEND):
+      command_rtn = sm_->unsuspend();
+      break;
+
+    default:
+      command_valid = false;
+  }
+  return command_valid && command_rtn;
+}
+
+bool PackmlRos::eventGuard(const int& event_id)
+{
+  switch(event_id)
+  {
+    case static_cast<int>(packml_sm::EventsEnum::STATE_COMPLETE):
+      sm_->triggerEvent(packml_sm::state_complete_event());
+      break;
+    case static_cast<int>(packml_sm::EventsEnum::HOLD):
+      sm_->triggerEvent(packml_sm::hold_event());
+      break;
+    case static_cast<int>(packml_sm::EventsEnum::UNHOLD):
+      sm_->triggerEvent(packml_sm::unhold_event());
+      break;
+    case static_cast<int>(packml_sm::EventsEnum::SUSPEND):
+      sm_->triggerEvent(packml_sm::suspend_event());
+      break;
+    case static_cast<int>(packml_sm::EventsEnum::UNSUSPEND):
+      sm_->triggerEvent(packml_sm::unsuspend_event());
+      break;
+    case static_cast<int>(packml_sm::EventsEnum::RESET):
+      sm_->triggerEvent(packml_sm::reset_event());
+      break;
+    case static_cast<int>(packml_sm::EventsEnum::CLEAR):
+      sm_->triggerEvent(packml_sm::clear_event());
+      break;
+    default:
+      ROS_ERROR_STREAM_NAMED("packml", "Event request called with invalid event_id: " << event_id);
+      return false;
+  }
+  return true;
+}
+
+bool PackmlRos::incStat(const int& metric, const double& step)
+{
+  bool result = true;
+  switch(metric)
+  {
+    case 0:
+      //do nothing
+      break;
+    case 1:
+      sm_->incrementSuccessCount();
+      break;
+    case 2:
+      sm_->incrementFailureCount();
+      break;
+    default:
+      if(metric >= 1000 && metric <= 1999)
```

Comment:
These seems kind of like magic numbers. Should they be params or at least class properties?

Reply:
Agreed. They're now in common as an enum.